core/src/main/java/org/frankframework/jdbc/datasource/ObjectCreator.java

```diff
@@ -117,33 +117,46 @@ private Properties getConnectionProperties(FrankResource resource, Properties en
 				}
 			}
 		}
-		CredentialFactory cf = getCredentials(resource);
-		if(StringUtils.isNotEmpty(cf.getUsername())) {
-			mergedProps.setProperty("user", cf.getUsername());
-		}
-		if(StringUtils.isNotEmpty(cf.getPassword())) {
-			mergedProps.setProperty("password", cf.getPassword());
+
+		String alias = substituteVariables(resource.getAuthalias());
+		String username = substituteVariables(resource.getUsername());
+		String password = substituteVariables(resource.getPassword());
+
+		// if alias is null, use username and password directly
+		if (StringUtils.isEmpty(alias)) {
+			setPropertyIfNotEmpty(mergedProps, "user", username);
+			setPropertyIfNotEmpty(mergedProps, "password", password);
+		} else {
```

Comment:
Ik vraag mij toch af of dit nodig is, ik zou verwachten dat dit generiek gebeurd en niet overal aangepast zou hoeven te worden.